### PR TITLE
UDP FOCS

### PR DESCRIPTION
**Pull request by geoffthemedio**
Allows UDP messages to server to provide more information than just whether the server can host a game. Simple context-free FOCS queries are parsed and evaluated and returned by UDP message to the sender.

### CODE REVIEW

**Comment by geoffthemedio** [comment]
Not really but perhaps somewhat relevant to https://github.com/freeorion/freeorion/issues/2497

**Comment by o01eg** [comment]
Is it safe to execute FOCS scripts from unknown source?

**Comment by o01eg** [comment]
Also it could lead to the game disclosure.

**Comment by geoffthemedio** [comment]
The usable "scripts" are quite limited... Just single free variables of int, double, or string. This means stuff that doesn't reference any object... for example `CurrentTurn`, `GalaxySeed`, or `UniverseWidth`. No conditions or effects. No `Source.Population` or `SetPopulation`. No statistics or arithmetic.

That said, I can add a bunch more stuff within or outside the FOCS portion, to provide additional info that would be useful. Ideally stuff that just needs a single word to describe, ie. no parameters, and can be returned as a string, either by the result naturally being a string, or by composition of multiple strings, or casting a number to a string as is already done for the non-string free variables after evaluating them.

**Comment by o01eg** [comment]
Ok, then it looks good. Could multiple variables be requested with a single request?

**Comment by geoffthemedio** [comment]
> Could multiple variables be requested with a single request?

In principle, something could be set up to do a more complicated parse of the input, but why not just send multiple queries?

**Comment by o01eg** [comment]
> In principle, something could be set up to do a more complicated parse of the input, but why not just send multiple queries?

I suppose single query is a simple for external application. But this PR is fine although.

File: parse/Parse.h [hunk 1]
```diff
@@ -72,12 +72,16 @@ namespace parse {
...
     FO_PARSE_API void file_substitution(std::string& text, const boost::filesystem::path& file_search_path);
-    FO_PARSE_API void process_include_substitutions(std::string& text, 
-                                                    const boost::filesystem::path& file_search_path, 
+    FO_PARSE_API void process_include_substitutions(std::string& text,
+                                                    const boost::filesystem::path& file_search_path,
                                                     std::set<boost::filesystem::path>& files_included);
+
+    FO_PARSE_API bool int_free_variable(std::string& text);
+    FO_PARSE_API bool double_free_variable(std::string& text);
+    FO_PARSE_API bool string_free_variable(std::string& text);
```
Comment (Morlic-fo) [inline parse/Parse.h:84]: I think the compilation units where these functions are defined do not to include ```parse.h``` and thus will never see the FO_PARSE_API declaration. Subsequently, the function visibility is still set to hidden and the linker won't find them.

Comment (geoffthemedio) [reply parse/Parse.h:84]: Looks like that was it, thanks.

**Pull request merged**